invenio_records_resources/records/api.py

```diff
@@ -63,11 +63,11 @@ def get_by_key(cls, record_id, key):
                 return cls(obj.data, model=obj)
 
     @classmethod
-    def list_by_record(cls, record_id, with_deleted=False):
+    def list_by_record(cls, record_id, include_deleted=False):
```

Comment:
unfortunately this might cause issues.
the with_deleted is coming from invenio_records and I think we are now hiding this functionality

Reply:
see also (got it while testing):
<img width="1715" alt="Screenshot 2023-09-25 at 17 02 04" src="https://github.com/inveniosoftware/invenio-records-resources/assets/38131488/5725770a-9970-4c35-a27c-b8d7c5f5c822">

Reply:
The error seems to be due to missing the changes from invenio-app-rdm, probably you are missing the changes of the [following PR ](https://github.com/inveniosoftware/invenio-app-rdm/pull/2434). If not, please describe how can I reproduce it and I will fix it.